### PR TITLE
A gallery example for the contour method

### LINKED ISSUE #1340: Add gallery/tutorial for Figure.contour()
The [contour](https://www.pygmt.org/dev/api/generated/pygmt.Figure.contour.html) method currently does not have any gallery plots or tutorial examples that show how it is used. 

The options to fix this include 1) adding a new gallery example for contouring table data or 2) adding a section to the [create a map with contour lines](https://www.pygmt.org/dev/tutorials/contour_map.html) tutorial that shows how to create a contour map from table data rather than gridded data.

Details for adding gallery plots and tutorial examples are in the [contributing guide](https://www.pygmt.org/dev/contributing.html).

### PR DESCRIPTION
**Pull request by yohaimagen**
**Description of proposed changes**

In response to issue [#1340](https://github.com/GenericMappingTools/pygmt/issues/1340) adding a gallery example for the contour method.  

Fixes #1340

**Preview** at https://pygmt-git-fork-yohaimagen-contour-gallery-example-gmt.vercel.app/gallery/images/contours.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:3]:
```suggestion
Contour (tabular data)
----------------------
```
I am recommending this change because it is likely that we'll add another gallery example for contour plots of gridded data.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:4]:
```suggestion
The :meth:`pygmt.Figure.contour` method can plot contour lines from a table of
points. Here, we supply the data as :class:`numpy.ndarray` with the ``x``, ``y``,
and ``z`` parameters and draw the contours using a 0.5p pen.
```
I have suggested adding a second sentence with a bit more detail about the method.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:12]:
```suggestion
Z = X ** 2 + Y ** 2
```
This is just a formatting suggestion that was caught by our linter.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:23]:
```suggestion
    pen="0.5p",
```
Specifying the units is helpful here ("p" for points) with a slightly thinned pen width.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:25]:
```suggestion
fig.show()

```
Just another suggestion based on the linter to have a blank line at the end of the module.

**Review by meghanrjones — COMMENTED** [review]
Great work @yohaimagen; thanks for addressing this issue! I have made a few suggestions on this pull request. If you agree with the suggestions, you can commit the changes using these steps:

1. Click on the 'Files changed' tab in the pull request
    image
2. For each comment, you can click 'commit suggestion' in order to commit the suggestions one at a time or click 'add suggestion to batch', which allows you to commit multiple suggestions at once. If you plan to accept multiple suggestions, I recommend using the 'add suggestion to batch' option.
    image
2a. If you clicked 'commit suggestion', you'll see a pop-up where you can enter a descriptive commit message and then click 'commit changes'.
2b. If you clicked 'add suggestion to batch', after reviewing all the suggestions, you can click 'commit suggestions' near the top of the 'files changed' tab. This will similarly prefill a commit message and allow you to commit the changes.

Feel free to reach out if you have any questions!

**Comment by yohaimagen** [comment]
Sorry I didn't saw your suggestions before adding some of my own, I incorporate some of your suggestions into mine.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:10]:
```suggestion
annotation on the contours respectively.

```
In sphinx with ReStructuredText (rst), a blank line is used to make sure that the next sentence is displayed on a new line.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:5]:
```suggestion
The data to the triangulation can be provided in one of three options:
```

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:8]:
```suggestion

#. ``x``, ``y``, ``z`` 1d :class:`numpy.ndarray` data columns.
#. ``data`` 2d :class:`numpy.ndarray` data matrix with 3 columns corresponding
   to ``x``, ``y``, ``z``.
#. ``data`` path string to a file containing the ``x``, ``y``, ``z`` in a
   tabular format.

```
This suggestion applies ReStructuredText formatting to the list.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:9]:
```suggestion
The parameters ``levels`` and ``annotation`` set the intervals of the contours and the
```

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:12]:
```suggestion
and ``z`` parameters and draw the contours using a 0.5p pen with contours every 10 ``z`` values and
```
I think it's a bit easier to see the difference between the contour spacing and annotation spacing if the contours are slightly farther apart, so I am suggestive a contour level of 10 rather than 5.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:37]:
```suggestion
    levels=10,
```

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:20]:
```suggestion
# build the contours underlying data with the function z = x^2 + y^2
```

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:32]:
```suggestion
    # pass the data as 3 1d data columns
```
Small recommendation to have the comments in the same form.

**Review by meghanrjones — COMMENTED** [review]
No worries, your revisions are great! I have made a set of small comments, mostly related to sphinx formatting, which can be really particular. I also recommended to increase the contour spacing from 5 to 10 in order to make it easier to see the difference between the contour spacing and the annotation spacing.

File: examples/gallery/lines/contours.py
Comment (meghanrjones) [inline examples/gallery/lines/contours.py:36]:
```suggestion
    # set the contours z values intervals to 10
```
One more suggestion for if you agree with changing contour interval from 5 to 10.

**Comment by yohaimagen** [comment]
Great, I committed all of your suggestions @meghanrjones. I think that this gallery example is done :)

**Comment by yohaimagen** [comment]
One last thing, maybe it will be better to put this example under "Images, contours, and fields" than under "Lines and vectors".

**Review by meghanrjones — APPROVED** [review]
Great work!

We usually leave approved pull requests open for ~1 day or until we get a second approval before merging, just in case other maintainers see anything major that needs adjusting.

File: examples/gallery/images/contours.py
Comment (willschlitzer) [inline examples/gallery/images/contours.py:5]:
```suggestion
The data for the triangulation can be provided using one of three methods:
```
Optional fix/personal preference, but I think this version reads better.

**Comment by meghanrjones** [comment]
Great work @yohaimagen, congratulations on merging the PR! :tada: We hope to see you around more 😄

**Comment by weiji14** [comment]
Cheers @yohaimagen! Your gallery example is now live at https://www.pygmt.org/dev/gallery/index.html#images-contours-and-fields. Well done! :partying_face:

image

**Pull request merged**
